pkg/pulumiyaml/run.go

```diff
@@ -1345,7 +1437,9 @@ func (e *programEvaluator) registerResource(kvp resourceNode) (lateboundResource
 
 	// Now register the resulting resource with the engine.
 	if isComponent {
-		err = e.pulumiCtx.RegisterRemoteComponentResource(string(typ), resourceName, untypedArgs(props), res, opts...)
+		typ := tokens.Type(typ)
+		packageRef := e.packageRefs[typ.Package()]
```

Comment:
Any cases where this lookup could fail? Same for below.

Reply:
It can, but the zero value is valid here

Reply:
Cool :ok_hand: